File: pset1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

// Prototypes
void print_row(int n, char s);

int main(void)
{
    // Prompt the user for a positive height
    int height = 0;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1);

    // Draw right-aligned pyramid
    for (int i = 0; i < height; i++)
    {
        // Draw a row of white spaces
        print_row(height - i - 2, ' ');

        // Draw a row of bricks of size i
        print_row(i, '#');

        // new row
        printf("\n");
    }   
}

// print a row of specific symbol
void print_row(int n, char s)
{
    for (int i = 0; i <= n; i++)
    {
        printf("%c", s);
    }
}

File: pset6/sentimental-mario-less/mario.py
def main():
    # Prompt user for a positive integer between 1 & 8
    height = 0

    while True:
        try:
            height = int(input("Height: "))
        except ValueError:
            height = 0
            pass

        if (height > 0) and (height <= 8):
            break
    
    # Draw right-aligned pyramid
    draw_pyramid(height)


def draw_pyramid(h):
    for i in range(h):
        # Draw spaces
        print(" " * (h - i - 1), end="")

        # Draw bricks
        print("#" * (i + 1), end="")

        # New row
        print()


if __name__ == "__main__":
    main()